lib/resque/scheduler/lock/resilient.rb

```diff
@@ -32,6 +33,21 @@ def timeout=(seconds)
 
         private
 
+        def evalsha(script, keys:, argv:, refresh: false, final_try: false)
+          sha_method_name = "#{script}_sha"
+          Resque.redis.evalsha(
+            send(sha_method_name, refresh),
+            keys: keys,
+            argv: argv
+          )
+        rescue Redis::CommandError => e
+          if e.message =~ /NOSCRIPT/ && !final_try
+            evalsha(script, keys: keys, argv: argv, refresh: true, final_try: true)
```

Comment:
I think instead of calling the method recursively (and using this `final_try` flag), it might be cleaner to use the `retry` keyword (and set `refresh = true` before calling it).